src/map/status.cpp

```diff
@@ -11298,6 +11406,85 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 			val2 = 30; // SP Recovery rate bonus
 			break;
 
+		case SC_SUNSTANCE:
+			val2 = 2 + val1; // ATK Increase
+			tick = -1;
+			break;
+		case SC_LUNARSTANCE:
+			val2 = 2 + val1; // MaxHP Increase
+			tick = -1;
+			break;
+		case SC_STARSTANCE:
+			val2 = 4 + 2 * val1; // ASPD Increase
+			tick = -1;
+			break;
+		case SC_UNIVERSESTANCE:
+			val2 = 2 + val1; // All Stats Increase
+			tick = -1;
+			break;
+		case SC_NEWMOON:
+			val2 = 7; // Number of Regular Attacks Until Reveal
```

Comment:
About SC_NEWMOON have 3 bugs.

1.
Even this place set val2 = 7;
But when under attack 1 times,
SC_NEWMOON will be remove and the player should be visible.

It should be defense 7 times.

2.
Consume SP not work.
Should 1 sp / per second

3.
Use SJ_NEWMOONKICK second times.
SC_NEWMOON should be remove and the player should be visible.

Reply:
Do you have a video or something to show part 3? It would make sense but I just want to confirm. Rytech didn't code it like this.

Reply:
https://www.youtube.com/watch?v=mjUvbPuVRQY

3.50 - 3.60

Reply:
He appear after use Full moon kick again

Reply:
[youtube](https://youtu.be/lpog7jF3qaE?t=466)

at 7:46  he used SJ_NEWMOONKICK  first time.
at 7:50 he used second times.

And TWRO skill descript also talk about this.